From 7c805b5d4eca798a03ac56b40e53cb335e30c187 Mon Sep 17 00:00:00 2001
From: Vlad Orlov <monsta@inbox.ru>
Date: Mon, 15 Aug 2016 18:23:31 +0200
Subject: [PATCH] libcryptui: fix logic flaw in the prompt recipients dialog

The prompt recipients dialog has been broken in commit
https://github.com/GNOME/libcryptui/commit/cd74aa6bf810a5ce0935d2ec89d6db64dbbde24d#diff-f0ea8a1eef5386b0149314d2a1743e85L202

The patch fixes the logic there and makes seahorse-tool's
encrypt command work again. Therefore, seahorse plugins for both
Nautilus and Nemo should start working as well.

Bug-Debian: https://bugs.debian.org/770436

https://bugzilla.gnome.org/show_bug.cgi?id=769944

Reviewed-by: Stef Walter <stefw@gnome.org>
---
 libcryptui/cryptui.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libcryptui/cryptui.c b/libcryptui/cryptui.c
index 92f0d5d4..2bbfa409 100644
--- a/libcryptui/cryptui.c
+++ b/libcryptui/cryptui.c
@@ -211,7 +211,7 @@ cryptui_prompt_recipients_with_symmetric (CryptUIKeyset *keyset,
             *symmetric = cryptui_key_chooser_get_symmetric (chooser);
         }
         
-        if (symmetric != NULL && !*symmetric) {
+        if (symmetric == NULL || *symmetric == FALSE) {
             recipients = cryptui_key_chooser_get_recipients (chooser);
             keys = g_new0(gchar*, g_list_length (recipients) + 1);
             for (l = recipients, i = 0; l; l = g_list_next (l), i++)
-- 
GitLab

